Mathlib/AlgebraicTopology/SimplexCategory/GeneratorsRelations/NormalForms.lean

```diff
@@ -155,4 +151,288 @@ theorem simplicialInsert_isAdmissible (L : List ℕ) (hL : IsAdmissible (m + 1)
+    simp at hk
+    subst hk
+    haveI := h_rec _ _ hL.tail (by simp +arith) hj
+    rw [← this]
+    generalize_proofs u
+    generalize
+      ((SimplexCategory.Hom.toOrderHom (toSimplexCategory.map (standardσ L u))) ⟨j, hj⟩) = t
+    simp only [Fin.predAbove, toSimplexCategory_obj_mk, SimplexCategory.len_mk, a₀]
+    split_ifs with h₁ h₂ h₂
+    · simp
+    · exfalso; simp [a₀, Fin.lt_def] at h₁; omega
+    · exfalso; simp [a₀, Fin.lt_def] at h₁; omega
+    · simp
+
+/-- Performing a simplicial insert in a list is the same as composition on the right by the
+corresponding degeneracy operator. -/
+lemma standardσ_simplicialInsert (hL : IsAdmissible (m + 1) L) (j : ℕ) (hj : j < m + 1) :
+    standardσ (m₂ := m) (simplicialInsert j L) rfl =
+    standardσ (m₂ := m + 1) L (by simp +arith [simplicialInsert_length]) ≫ σ j := by
+  induction L generalizing m j with
+  | nil => simp [standardσ, simplicialInsert]
+  | cons a L h_rec =>
+    simp only [List.length_cons, eqToHom_refl, simplicialInsert, Category.id_comp]
+    split_ifs <;> rename_i h <;> simp only [standardσ, eqToHom_trans_assoc, Category.assoc]
+    haveI := h_rec hL.tail (j + 1) (by simpa)
+    generalize_proofs h _ _ h' at this ⊢
+    haveI := standardσ
+      (m₁ := m + (if j < a then j :: a :: L else a :: simplicialInsert (j + 1) L).length) []
+        (by aesop) ≫= this
+    rw [standardσ_comp_standardσ] at this
+    simp only [List.append_eq, List.append_nil] at this
+    rw [this, reassoc_of% standardσ_comp_standardσ]
+    simp only [List.append_eq, List.append_nil, Category.assoc]
+    slice_rhs 2 4 =>
+      equals σ ↑(j + 1) ≫ σ a =>
+        have ha : a < m + 1 := by omega
+        haveI := σ_comp_σ_nat a j ha hj (by omega)
+        generalize_proofs p p' at this
+        -- We have to get rid of the natCasts...
+        have ha₁ : (⟨a, p⟩ : Fin <| m + 1 + 1) = ↑a := by ext; simp [Nat.mod_eq_of_lt p]
+        have ha₂ : (⟨a, ha⟩ : Fin <| m + 1) = ↑a := by ext; simp [Nat.mod_eq_of_lt ha]
+        have hj₁ : (⟨j + 1, p'⟩ : Fin <| m + 1 + 1) = ↑(j + 1) := by ext; simp [Nat.mod_eq_of_lt p']
+        have hj₂ : (⟨j, hj⟩ : Fin <| m + 1) = ↑j := by ext; simp [Nat.mod_eq_of_lt hj]
+        rwa [← ha₁, ← ha₂, ← hj₁, ← hj₂]
+
+/-- Using `standardσ_simplicialInsert`, we can prove that every morphism satisfying `P_σ` is equal
+to some `standardσ` for some admissible list of indices. Because morphisms of the form `standardσ`
+have a rather  constrained sources and targets, we have again to splice in some `eqToHom`'s to make
+everything work. -/
+theorem exists_normal_form_P_σ {x y : SimplexCategoryGenRel} (f : x ⟶ y) (hf : P_σ f) :
+    ∃ L : List ℕ,
+    ∃ m : ℕ, ∃ b : ℕ,
+    ∃ h₁ : mk m = y, ∃ h₂ : x = mk (m + b), ∃ h : L.length = b,
+    IsAdmissible m L ∧ f = standardσ L (by rw [h, h₁.symm, h₂]; rfl) := by
+  induction hf with
+  | id n =>
+    use [], n.len, 0, rfl, rfl, rfl, IsAdmissible.nil _
+    simp only [standardσ, eqToHom_refl]
+    rfl
```

Comment:
```suggestion
    simp
    rfl
```

Reply:
Won’t the linter complain about the non-terminal simp?

Reply:
(actually `rfl` without simp before works, so that’s a non-problem).